File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Polygonator - Menon e Dalla Via/Java/src/Model/Polygons/Rhombus.java
package Model.Polygons;

import Model.Point;

import java.util.List;

public class Rhombus extends Kite {
    public Rhombus(final List<Point> vertices) throws Exception {
        super(vertices);
    }

    /**
     * Returns whether a polygon is a rhombus or not
     */
    public static boolean canBeCreatedWith(final Polygon polygon) throws Exception {
        return Parallelogram.canBeCreatedWith(polygon) && Kite.canBeCreatedWith(polygon);
    }

    @Override
    public double getPerimeter() throws Exception {
        return getBase() * 4;
    }

    @Override
    public String toString() {
        return "Rhombus";
    }
}

File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Polygonator - Menon e Dalla Via/C++/Model/Polygons/Rhombus.cpp
#include "Rhombus.h"

Rhombus::Rhombus(const list<Point> &vertices) : Quadrilateral(vertices), Parallelogram(vertices), Kite(vertices) {}

bool Rhombus::canBeCreatedWith(const Polygon &polygon) {
    return Parallelogram::canBeCreatedWith(polygon) && Kite::canBeCreatedWith(polygon);
}

double Rhombus::getArea() const {
    return Kite::getArea();
}

double Rhombus::getPerimeter() const {
    return getBase() * 4;
}

string Rhombus::getName() const {
    return "Rhombus";
}
